src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -336,29 +387,39 @@ public virtual bool TryHandleEscape()
             if (ExpansionSession != null)
             {
                 ExpansionSession.EndCurrentExpansion(fLeaveCaret: 1);
-                ExpansionSession = null;
+                _state.Clear(forceClearSymbolInformation: true);
                 return true;
             }
 
             return false;
         }
 
         public virtual bool TryHandleReturn()
+        {
+            return CommitSnippet(leaveCaret: false);
+        }
+
+        public bool CommitSnippet(bool leaveCaret)
         {
             if (ExpansionSession != null)
             {
-                // Only move the caret if the enter was hit within the snippet fields.
-                var hitWithinField = VSConstants.S_OK == ExpansionSession.GoToNextExpansionField(fCommitIfLast: 0);
-                ExpansionSession.EndCurrentExpansion(fLeaveCaret: hitWithinField ? 0 : 1);
-                ExpansionSession = null;
+                if (!leaveCaret)
+                {
+                    // Only move the caret if the enter was hit within the snippet fields.
```

Comment:
not quite certain i understand this.  if the user hits 'return', i would have previously guessed it just commits the snippet.  but now it also moves the caret?  if so, consider doc'ing to where.

Reply:
This is part of the previous implementation right?

Reply:
➡️ I'm leaving the grandfathered code here, but to answer the question the caret moves to the location where `$end$` appeared in the snippet text, otherwise to the end of the original snippet text.